src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -78,7 +103,78 @@
       final String symbol,
       final String range) {
     log.info("Retrieving historical price data for Symbol: {}, range: {}", symbol, range);
-    return iexClientCloud.getHistoricalPrice(symbol, range);
+
+    // Declare Variables
+    List<IexHistoricalPrice> historicalPrices;
+    List<IexHistoricalPriceEntity> historicalEntities;
+    LocalDate end = LocalDate.now();
+    LocalDate start;
+
+    // Determine Range Boundaries
+    switch (range.toUpperCase()) {
+      case "MAX":
+        start = end.minusYears(15);
+        log.info(start.toString());
+        break;
+      case "5Y":
+        start = end.minusYears(5);
+        log.info(start.toString());
+        break;
+      case "2Y":
+        start = end.minusYears(2);
+        log.info(start.toString());
+        break;
+      case "1Y":
+        start = end.minusYears(1);
+        log.info(start.toString());
+        break;
+      case "YTD":
+        start = end.withDayOfYear(1);
+        log.info(start.toString());
+        break;
+      case "6M":
+        start = end.minusMonths(6);
+        log.info(start.toString());
+        break;
+      case "3M":
+        start = end.minusMonths(3);
+        log.info(start.toString());
+        break;
+      case "1M":
+        start = end.minusMonths(1);
+        log.info(start.toString());
+        break;
+      case "5D":
+        start = end.minusDays(5);
+        log.info(start.toString());
+        break;
+      default:
+        start = end;
+        log.info("Range entry does not match possible values.");
+    }
+
+    // Check database for symbols
+    /*
+      I need a more accurate way to test if the database has the symbols rather than isEmpty().
+      If I get a specific date and then try to run this method, if that date was within the last
+       30 days, it'll just return that.  How can I do this?
```

Comment:
It seems like what you'd want to do is check to see how many dates you would expect to get returned if they were all in the database. For example, if I wanted everything in the last month `1m` and only 1 result (day) was returned by the database, then I'd know my database didn't have everything cached and I should just make the full call to the IEX API and save it all to my H2 database